tests/hardhat/Fork/diamondTest.ts

```diff
@@ -0,0 +1,414 @@
+import { smock } from "@defi-wonderland/smock";
+import { impersonateAccount } from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import chai from "chai";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers, network } from "hardhat";
+
+import { VBep20 } from "../../../typechain";
+import { deployDiamond } from "../Comptroller/Diamond/scripts/deploy";
+
+const { expect } = chai;
+chai.use(smock.matchers);
+
+const Owner = "0x939bd8d64c0a9583a7dcea9933f7b21697ab6396";
+const UNITROLLER = "0xfD36E2c2a6789Db23113685031d7F16329158384";
+const zeroAddr = "0x0000000000000000000000000000000000000000";
+const VBUSD = "0x95c78222B3D6e262426483D42CfA53685A67Ab9D";
+const VUSDT = "0xfD5840Cd36d94D7229439859C0112a4185BC0255";
+
+let owner,
+  unitroller,
+  diamondProxy,
+  // layout variables
+  oracle,
+  maxAssets,
+  closeFactorMantissa,
+  liquidationIncentiveMantissa,
+  allMarkets,
+  venusRate,
+  venusSpeeds,
+  venusSupplyState,
+  venusBorrowState,
+  venusAccrued,
+  vaiMintRate,
+  vaiController,
+  mintedVAIs,
+  mintVAIGuardianPaused,
+  repayVAIGuardianPaused,
+  protocolPaused,
+  venusVAIVaultRate,
+  vaiVaultAddress,
+  releaseStartBlock,
+  minReleaseAmount,
+  treasuryGuardian,
+  treasuryAddress,
+  treasuryPercent,
+  liquidatorContract,
+  comptrollerLens,
+  market,
+  venusSupplierIndex,
+  venusBorrowerIndex;
+
+const initMainnetUser = async (user: string) => {
+  await impersonateAccount(user);
+  return ethers.getSigner(user);
+};
+
+export async function setForkBlock(blockNumber: number) {
+  await network.provider.request({
+    method: "hardhat_reset",
+    params: [
+      {
+        forking: {
+          jsonRpcUrl: process.env.BSC_ARCHIVE_NODE,
```

Comment:
I think we are using the environment variable `BSC_ARCHIVE_NODE_URL` in the rest of the codebase.

```suggestion
          jsonRpcUrl: process.env.BSC_ARCHIVE_NODE_URL,
```